lib/http_aws_sigv4.c

```diff
@@ -409,6 +409,11 @@ static int compare_func(const void *a, const void *b)
 {
   const struct pair *aa = a;
   const struct pair *bb = b;
+  /* If one element is empty, the other is always sorted higher */
+  if(aa->len == 0)
+    return -1;
+  if(bb->len == 0)
+    return 1;
```

Comment:
For a second I thought this might need a special case for when both lengths are zero, but I doubt that is true. This code should make the sorting stable and working nonetheless.